### ISSUE TITLE
Sorting enum values alphabetically

### ISSUE THREAD

**Issue by schjonhaug** [nicklockwood/SwiftFormat#623]
It would be neat if SwiftFormat could sort enum values alphabetically, so that

```
enum Activity {
    case singing
    case running
    case talking
    case bored
}
```

would become

```
enum Activity {
    case bored
    case running
    case singing
    case talking
}
```

**Comment by nicklockwood** [comment]
This is doable, but it would have to be disable by default. I think it's safe in the simple case but for enums with `Int` type or that conformed to `CaseIterable` it would potentially change program behavior.

**Comment by schjonhaug** [comment]
Could it safely be enabled by default for simple cases?

**Comment by nicklockwood** [comment]
@schjonhaug it could, but I don't know if most users would necessarily want it.

**Comment by schjonhaug** [comment]
For small enums like in my example, probably not. But for larger ones, it's hard to scan the list when it's not sorted.

**Comment by VinceBurn** [comment]
This is not a feature that I would like to be on by default, since it has the potential to break code like @nicklockwood mention.

My personal feeling is that I'm not sure this should be in a code formatter, since it looks to me more like an adhoc thing that you may want on some case, but not on other.
It feel that this is something that should be left at developer judgment, 
unless the solution is to put a `swiftformat` comment before the enum to say "sorted" to indicate that this specific case should be sorted.

Other option would be to use an Xcode extension dedicated to sorting like "xcsort" (disclaimer, I haven't try it, but its available on the App Store), but that is not automatic.

**Comment by PompeiaPaetenari** [comment]
If this were going through, it needs to also apply to the switch statements over said enum, or similar.

Also, I think it needs to be configurable.  Perhaps as simple as a comment in the code or something to not do certain enums, but do all the rest of them.  In our code, for example, we sort all enums alphabetically except for the `CaseIterable`, as per Nick, and ones that are
```swift
enum result {
case success;
case failure;
}
```
 where we want to have it in that order.

**Comment by nicklockwood** [comment]
> For small enums like in my example, probably not. But for larger ones, it's hard to scan the list when it's not sorted.

So maybe it would make sense to apply it only to enums above a certain number of items?

**Comment by PompeiaPaetenari** [comment]
> > For small enums like in my example, probably not. But for larger ones, it's hard to scan the list when it's not sorted.
> 
> So maybe it would make sense to apply it only to enums above a certain number of items?

I agree to this with a caveat:  `--enumsortmin 5` or similar config options.  I think picking an arbitrary number is not going to be perfect for everyone.

**Comment by PompeiaPaetenari** [comment]
Hi Nick,

Just ran into a fairly large enum in our code that I wanted to sort and saw SwiftFormat not helping me do that... searched and found this old topic explaining why.  We do generally prefer to have enums and their switch statements sorted in our code, and a tool that a colleague made has not worked since Xcode 13, so curious if there were any developments on this effort?

Especially given we do have the success with `sortedSwitchCases` and have that rule enabled. :D

**Comment by calda** [comment]
You can add a `// swiftformat:sort` directive above the enum and its cases will be sorted automatically:

```swift
// swiftformat:sort
enum Activity {
    case bored
    case running
    case singing
    case talking
}
```

**Comment by brzzdev** [comment]
I'd also really like this for all enums that aren't `Int` and `CaseIterable`

Or at least the ability to configure it for all enums with a certain name. E.g - `enum Action`
